Add function to get the largest cutoff radius

g-bauer opened the pull request:
This PR adds functionality to access the largest cutoff from all `pairs`. This is useful to get the limit of the smallest simulation cell.

Currently, only pair interactions are considered, no electrostatics.

Luthaf commented on src/core/src/energy/pairs.rs:
```diff
@@ -147,6 +147,11 @@ impl PairInteraction {
     }
+
+    /// Return the cutoff radius
+    pub fn get_cutoff(&self) -> f64 {
```
As this is a public function, can you add an example here? See the other functions or Rust standard library for example of `examples` 😄 The example should use this function and have at least one `assert_eq!` for checking the function result.

Luthaf commented on src/core/src/sys/interactions.rs:
```diff
@@ -185,6 +185,20 @@ impl Interactions {
+    /// about which interaction belongs to which pair of particles.
+    pub fn all_pairs(&self) -> Vec<&PairInteraction> {
+        let res: Vec<&PairInteraction> = self.pairs
+            .values() // value is a Vec<&PairInteraction>
```
`Vec<Vec<&PairInteractions>>`

g-bauer replied:
I am not sure here. I thought that the value of the tree is a `Vec<PairInteraction>` looking at the definition in the struct. `flat_map` produces an iterator over the elements, i.e. `PairInteraction`s. This is done for every entry in the tree and finally everything is collected into a single vector. Did I get this wrong?

Luthaf commented on src/core/src/sys/interactions.rs:
```diff
@@ -185,6 +185,20 @@ impl Interactions {
+            // Vec<&PairInteraction>
+            .flat_map(|i| i)
+            .collect();
+        res
```
I believe this could be written

```rust
pub fn all_pairs(&self) -> Vec<&PairInteraction> {
    self.pairs.values() // value is a Vec<Vec<&PairInteraction>>
                        // flatten the vector -> return an iterator over
                        // Vec<&PairInteraction>
              .flat_map(|i| i)
              .collect()
}
```

Luthaf commented on src/core/src/sys/systems.rs:
```diff
@@ -465,6 +464,19 @@ impl System {
+
+    /// Return the largest cutoff radius from pair interactions
+    /// excluding electrostatic interactions.
+    pub fn largest_cutoff(&self) -> f64 {
```
I am not comfortable with having this as a `System` function. It feels like it belongs to `Interactions` instead, and should be accessed by `system.interactions().largest_cutoff()`

`System` already have too much functions =/

Aside, how often do you plan to use this ? If it is only for `Resize`, it could just be a piece of code inside resize, that we can refactor in a public function later if needed.

g-bauer commented:
I updated the comments and added an example for `get_cutoff`.

Also, I agree that a `fn largest_cutoff` is not needed since it will not be called too often and therefore it should be computed explicitly for each respective case. Sadly, the name of the PR does not capture the actual changes anymore.

Luthaf commented:
> Sadly, the name of the PR does not capture the actual changes anymore. 

This is not too important, and you can change the PR name !
